src/tagmanager.cpp

```diff
@@ -1,15 +1,22 @@
 #include "tagmanager.h"
 
+#include <cstddef>
 #include <memory>
+#include <ostream>
+#include <stdexcept>
 
+#include "attribute_.h"
```

Comment:
In principle, whenever `attribute_.h` is included in a cpp-file, then something is wrong (except for `rules.cpp`, where it's fine).